### DISCUSSION TITLE
Does anyone know if there is a sorts of node-id property/attribute?

### DISCUSSION THREAD

**Discussion by dgsus** [deso-protocol/dips#36] [Q&A] [answered]
Is there any attribute, that exists or is technically possible to be included, on any given transaction that would indicate where that transaction originated? Not a identifiable attribute such as IP, but pseudonymous. Perhaps a hash derived from node hardware/software? A simple use case would be to cluster transactions from a certain node, for forensics on possible bad actors.

Does anyone see this as viable or needed?

Cheers!

**Comment by bloated-devish** [comment]
Can you check member.cash's post today. They used ```ExtraData``` to implement user block in the chain.

**Reply by tijno** [reply]
Which one ? I cant see it

**Reply by bloated-devish** [reply]
https://bitclout.com/posts/8a7fbaf7440b9def365a682dafcd92d130532cc5a78772b5f5d14f4453701bc2

Check this

**Comment by dgsus** [comment, marked as answer]
@maebeam @lazynina perhaps you can offer some insight here? 
thx in advance!

**Reply by maebeam** [reply]
Today there is not. If you'd like, create a CIP discussion that proposes adding this field

**Reply by dgsus** [reply]
cheers!

**Reply by bloated-devish** [reply]
qq - @maebeam Are you thinking more on the lines of how you are plannign to do verification in the future - where there a link between node and post or do you plan to do something more tighter like allow nodes to encrypt transaction with their private key - so that the encrypted transaction can be decrypted in the chain only using the public key of the node, thus verifying that the transaction came from a particular node?

**Comment by sull** [comment]
I remember discussing this somewhere early on as to why this would be useful. Primarily, I was thinking about being able to track which node a username/public address was created.

Something like an enode URI (ethereum) combined with a domain name perhaps and the first new block transaction id the node was involved in creating. And include this hashed ID in all transactions, allowing the node to track itself while still being unidentified by the network at large. 

However, "forensics on possible bad actors" use case would be for private investigation purposes and probably of limited use since once revealing your node's association with an incident would also reveal your node's ID thus losing the pseudonymity. 

I may be looking at this the wrong way though.

**Reply by dgsus** [reply]
I will open a cip around this. We can discuss further under it.

Cheers Sull.
